### PR TITLE
add Partition Array Into Three Parts With Equal Sum

### PR DESCRIPTION
**Pull request by tkkawa**
Please check my code.

### CODE REVIEW

File: 1013_partition-array-into-three-parts-with-equal-sum.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#n = arr.length
+# time = O(n)
+# space = O(1)
+class Solution:
+    def canThreePartsEqualSum(self, arr: List[int]) -> bool:
+        sub_sum = sum(arr) / 3
```
Comment (ryuji0123) [inline 1013_partition-array-into-three-parts-with-equal-sum.py:6]: to make sure sub_sum is int, write like:

```py
sub_sum = sum(arr) // 3
```

File: 1013_partition-array-into-three-parts-with-equal-sum.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        sub = 0
+        a = 0
+        for i in range(len(arr)):
```
Comment (ryuji0123) [inline 1013_partition-array-into-three-parts-with-equal-sum.py:9]: if you don't update elements in arr and just add them to sub, you should write like:

```py
for num in arr:
    sub += num
```

File: 1013_partition-array-into-three-parts-with-equal-sum.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+            sub += arr[i]
+            if sub == sub_sum:
+                sub = 0
+                a += 1
+        if a > 2:
```
Comment (ryuji0123) [inline 1013_partition-array-into-three-parts-with-equal-sum.py:14]:
```py
if 2 < a:
```

Comment (ryuji0123) [reply 1013_partition-array-into-three-parts-with-equal-sum.py:14]: Also, you should return True or False more clearly

```py
return 2 < a
```

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
Very close

File: 1013_partition-array-into-three-parts-with-equal-sum.py (hunk 1 above)
Comment (ryuji0123) [reply 1013_partition-array-into-three-parts-with-equal-sum.py:6]: You miss this:
```py
sub_sum = sum(arr) // 3
```

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
see my comments

**Comment by tkkawa** [comment]
if sub_sum is int, 11th line is bad.
For example, when input is [1,1,1,1], output is true, but  true solution is false.

**Comment by ryuji0123** [comment]
use divmod

```py
sub_sum, m = divmod(sum(arr), 3)
if m != 0:
    return False
```

**Review by ryuji0123 — APPROVED** [review]
LGTM

**Pull request merged**